client/core/types.go

```diff
@@ -533,10 +533,11 @@ type dexAccount struct {
 	cert      []byte
 	dexPubKey *secp256k1.PublicKey
 
-	keyMtx  sync.RWMutex
-	encKey  []byte
-	privKey *secp256k1.PrivateKey
-	id      account.AccountID
+	keyMtx     sync.RWMutex
+	encKey     []byte
+	privKey    *secp256k1.PrivateKey
+	id         account.AccountID
+	hdKeyIndex uint32 // child index of account key, starts as 0 and increments for alternative keys for same DEX
```

Comment:
`index of last child of account keypair hd path`? 🤔

Reply:
`hdKeyIndex` will be back to zero each time Core is started.

Since I'm not inclined to add this to the database, I think it can just be a variable that is local to `registerHD`.  Specifically start with `var hdKeyIndex uint32` at the beginning of `registerHD`, pass `hdKeyIndex` to `setupCryptoV2` (removing the mutex locking and `HardenedKeyStart` check), if account is suspended, `hdKeyIndex++` right there instead of in `resetKeys`.  This also makes the [`len(a.encKey) > 0` check question](https://github.com/decred/dcrdex/pull/1210/commits/4322ac8029c5ce818c005244399a23b567381798#r708500390) a non-issue.